modules/platforms/dotnet/Apache.Ignite.Core/Binary/BinaryBasicNameMapper.cs

```diff
@@ -48,7 +48,7 @@ public class BinaryBasicNameMapper : IBinaryNameMapper
         /// <summary>
         /// Gets the type name.
         /// </summary>
-        public string GetTypeName(string name)
+        public virtual string GetTypeName(string name)
```

Comment:
@nizhikov please see the [dev list](http://apache-ignite-developers.2346864.n4.nabble.com/NET-naming-convention-RegisterSameJavaType-mode-td51386.html) - making those methods `virtual` does not seem to be a good idea.

Reply:
I reworked the solution according to your proposals on the dev-list.

`NamespaceToLower` and `JavaDomain` flags added to `BinaryBasicNameMapper`.
Please, take a look.